### PR TITLE
Telemetry MetricReportsDefinition Collections update

### PR DESCRIPTION
**Pull request by aahmed-2**
1) Refactor MetricReport in EventService.
2) Add CollectionTimeScope, CollectionFunction, CollectionDuration to MetricReportDefinition schema.

This PR requires a rebase of [Telemetry repo on upstream master for synchronization between bmcweb and telemetry binaries.](https://github.com/openbmc/telemetry/commits/master)
GHE-1020 srcrev bump: https://github.ibm.com/openbmc/openbmc/pull/1653 

NB:

1. Using ReportUpdate signal is no longer supported. Instead, Event Service monitors for PropertiesChanged signal from /xyz/openbmc_project/Telemetry/Reports path.  

Commits:
"Sync Telmetry service with EventService":
Change-Id: I2fc1841a6c9259a8bff30b34bddc0d4aabd41912

"Switched bmcweb to use new telemetry service API":
Change-Id: I2cd17069e3ea015c8f5571c29278f1d50536272a
https://gerrit.openbmc-project.xyz/c/openbmc/bmcweb/+/44270/25

### CODE REVIEW

**Comment by gtmills** [comment]
Nitpick: the link with the patchset used might be slightly better: e.g. https://gerrit.openbmc-project.xyz/c/openbmc/bmcweb/+/44270/24 ?

> This is update rolls back the "MetricsProperties" functionality under MetricReportDefinition.
"MetricProperties" can now be defined under "Metrics" along with CollectionTimeScope, CollectionFunction, and CollectionDuration.

Where did you take this "MetricsProperties" functionality from?

> Using ReportUpdate signal is no longer supported. Instead, Event Service monitors for PropertiesChanged signal from /xyz/openbmc_project/Telemetry/Reports path.

You did this change? 
so 1 and 2 are things you did on top of what you pulled in?

I kind of thought, the steps would be more like cherry pick first, followed by changes done by yourself
1. cherry pick https://gerrit.openbmc-project.xyz/c/openbmc/bmcweb/+/44270/24
2. cherry pick ...
3.  Took "MetricsProperties" functionality under MetricReportDefinition from an earlier patchset of 44270 and added back
4. ...

**Comment by aahmed-2** [comment]
None of this is my changes. 

Commit "Sync Telmetry service with EventService" is already merged upstream. We want it to make sure our MetricReportDefinition schema property ReportUpdate works with the new Telemetry repo master downstream.

**Comment by aahmed-2** [comment]
> This is update rolls back the "MetricsProperties" functionality under MetricReportDefinition.
> "MetricProperties" can now be defined under "Metrics" along with CollectionTimeScope, CollectionFunction, and CollectionDuration.
> 
> Where did you take this "MetricsProperties" functionality from?

I made a mistake MetricProperties was NOT previously supported as I had thought. MetricsProperties was only ever implemented as a sub-property of Metrics. The MetricsReportDefinition schema does have "MetricProperties" property defined in 2 places: 1) Under "Metrics", and 2) as a top-level property.

Screen Shot 2021-12-13 at 9 40 35 AM

File: redfish-core/lib/metric_report_definition.hpp [hunk 1]
```diff
@@ -203,31 +232,50 @@ inline bool getUserParameters(crow::Response& res, const crow::Request& req,
...
+                metricArgs.collectionTimeScope, "CollectionDuration",
+                collectionDurationStr))
         {
             return false;
         }
 
-        args.metrics.emplace_back(std::move(id), std::move(uris));
+        if (collectionDurationStr)
+        {
+            std::optional<std::chrono::milliseconds> duration =
```
Comment (patelabhishek9893) [inline redfish-core/lib/metric_report_definition.hpp:249]: why we created std::optinal?

**Review by patelabhishek9893 — APPROVED** [review]

**Review by patelabhishek9893 — APPROVED** [review]

**Comment by gtmills** [comment]
> This PR requires a rebase of Telemetry repo on upstream master for synchronization between bmcweb and telemetry binaries.

Where is status here?

**Review by gtmills — APPROVED** [review]
Seems fine, going to default to Ali and the Intel work upstream

**Pull request merged**